crates/currency/src/lib.rs

```diff
@@ -42,7 +83,9 @@ pub mod pallet {
     /// ## Configuration
     /// The pallet's configuration trait.
     #[pallet::config]
-    pub trait Config: frame_system::Config + orml_tokens::Config<Balance = BalanceOf<Self>> {
+    pub trait Config:
+        frame_system::Config + orml_tokens::Config<Balance = BalanceOf<Self>, CurrencyId = PrimitivesCurrencyId>
```

Comment:
Let's not use the concrete type:

```suggestion
        frame_system::Config + orml_tokens::Config<Balance = BalanceOf<Self>, CurrencyId = CurrencyId<Self>>
```

Reply:
This somehow triggers a compiler bug: https://stackoverflow.com/questions/66631251/why-do-i-get-an-overflow-evaluating-the-requirement-error-for-a-simple-trait-i
```
error[E0275]: overflow evaluating the requirement `<T as orml_tokens::Config>::CurrencyId == _`
  --> crates/currency/src/amount.rs:18:46
   |
18 | #[cfg_attr(feature = "testing-utils", derive(Copy))]
   |                                              ^^^^
   |
   = note: this error originates in the derive macro `Copy` (in Nightly builds, run with -Z macro-backtrace for more info)
   ```

Reply:
I think it's because `CurrencyId` is defined like this, so there might be a reference loop?
```rust
pub type CurrencyId<T> = <T as orml_tokens::Config>::CurrencyId;
```

Reply:
The reason I specified the concrete type in the first place was this error:
```rust
error[E0599]: no method named `is_lend_token` found for associated type `<T as orml_tokens::Config>::CurrencyId` in the current scope
  --> crates/currency/src/lib.rs:46:30
   |
46 |         if amount.currency().is_lend_token() && to.is_lend_token() {
   |                              ^^^^^^^^^^^^^ method not found in `<T as orml_tokens::Config>::CurrencyId`
   ```